internal/k8s/client.go

```diff
@@ -135,6 +136,8 @@ type Client interface {
 	OwnerFetcher() OwnerFetcher
 
 	ClusterHealth(ctx context.Context, verbose bool) (ClusterHealth, error)
+
+	ConnectionStatus() *v1alpha1.KubernetesClusterConnectionStatus
```

Comment:
Ok so in the context of the API `KubernetesClusterConnectionStatus` makes sense BUT here I wonder if the method should be something like `ConnectionConfig()`?

Reply:
good idea! done